src/main/java/com/chain4travel/cmbplugin/grpc/converter/MessageConverter.java

```diff
@@ -0,0 +1,101 @@
+
+
+package com.chain4travel.cmbplugin.grpc.converter;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import build.buf.gen.cmp.services.accommodation.v1alpha.AccommodationSearchRequest;
+import build.buf.gen.cmp.services.accommodation.v1alpha.AccommodationSearchResponse;
+import build.buf.gen.cmp.services.book.v1alpha.ValidationRequest;
+import build.buf.gen.cmp.services.book.v1alpha.ValidationResponse;
+import build.buf.gen.cmp.services.transport.v1alpha.TransportSearchRequest;
+import build.buf.gen.cmp.services.transport.v1alpha.TransportSearchResponse;
+
+
+public class MessageConverter {
+
+    public static AccommodationSearchRequest readAccommodationSearchRequest(InputStream inputStream) {
```

Comment:
Could that be done with meta programming in Java? In C++ with templates this could be a single method without the repeating everything except the type.

Just as a hint that we might want to think about having this converter as a template so that it's not needed to change/extend it every time a new request/response is added.

Reply:
I tried to use Java generics here but it did not work out really well.